Add newsletter post (#11054)
PostHog/posthog.com: contents/newsletter/good-taste-great-products.md

---
title: Good taste makes great products
date: 2025-03-06
author:
 - danilo-campos
featuredImage: >-
  https://res.cloudinary.com/dmukukwp6/image/upload/418950516_1ae2b12e_a571_48f6_a979_2c89deb7a7f9_c1c4010916.jpg
featuredImageType: full
tags:
  - Engineering
  - Product
crosspost:

>   - Product engineers

Reviewer: ```suggestion
  - Product engineers
  - Blog
```

---

> When we talk about how to be a good [product engineer](/blog/what-is-a-product-engineer), we often focus on how you need to ship fast, be customer obsessed, and [take extreme ownership of ideas](https://newsletter.posthog.com/p/hiring-and-managing-cracked-engineers). 

Reviewer: ```suggestion
When we talk about how to be a good [product engineer](/blog/what-is-a-product-engineer), we often focus on how you need to ship fast, be customer obsessed, and [take extreme ownership of ideas](/founders/cracked-manifesto). 
```

Minor nit I frequently forget myself, but we should link to website versions of newsletters on the website.

But one important quality is subjective: taste.

Taste means intuition for what makes a product feel great to use, and knowing how to evoke those feelings in your own work.

It’s these emotions that earn loyalty and convince others to share what you build.

No one is born with good taste. It’s something we cultivate over time through exposure to good (and bad) things.

In other words, taste is a muscle you can train.

* * *

> ##  **1. Understand and embrace affordances**

Author: ugh sorry to make you flag all those h2/3's, I thought I was being clever with my little converter

An **affordance** is the way an object communicates how it works. In _[The Design of Everyday Things](https://www.amazon.com/Design-Everyday-Things-Revised-Expanded/dp/0465050654)_, Don Norman provides the example of a door handle.

Doors you _pull_ should have a nice handle for gripping and yanking, while doors you _push_ should have a broad surface that shouts “you need to shove this.”

![](https://res.cloudinary.com/dmukukwp6/image/upload/aa0f8629_f6b0_4f30_8289_4f69e26d0110_1600x1600_2ebb3ba866.jpg)

> _Credit: [Scott Brody](https://commons.wikimedia.org/w/index.php?curid=55516950)_

Author: OHHH will that allow the caption element to also parse its containing elements?

> _Credit: [Scott Brody](https://commons.wikimedia.org/w/index.php?curid=55516950)_

Author: womp womp

<img width="658" alt="Screenshot 2025-03-24 at 4 14 36 PM" src="https://github.com/user-attachments/assets/f8d3a41f-a208-4d37-a230-5963e09c94de" />

> _Credit: [Scott Brody](https://commons.wikimedia.org/w/index.php?curid=55516950)_

Author: oh sweet, THAT does now work in mdx, thank youuuuuu

Here are a few more you still see today:

  *  **Buttons** – clickable regions – have outlines and often distinct colors

  *  **Text fields** have borders and sometimes inner shadows, along with placeholder text to provide examples of what you should enter

  * A **pop-up menu** often sports a small triangle, pointed downward, to suggest that clicking it will reveal additional options

You need to play with lots of software to stay abreast of affordances – it’s vital for learning the design vocabulary users will expect and recognize.

Make a list of all the affordances you observe in software you use frequently. Do the same for products that are similar to what you’re building and look for patterns. 

The more you do this, the more you’ll instinctively build them into your work, thus making your software feel effortless and familiar. 

> **Recommended reading:** [_Design is Storytelling_](https://www.amazon.com/Design-Storytelling-Ellen-Lupton/dp/194230319X) by Ellen Upton is a fun treatise on the power of narrative to help software communicate with users.

* * *

> ##  **2. Detect and eliminate obvious jank**

Reviewer 2: ```suggestion
##  2. Detect and eliminate obvious jank
```

You don’t need to be the most talented designer on earth to have taste.

Instead, you need a reliable sense of what is janky and what is nice, and the ability to nudge a product in the nicer direction.

People who encounter janky software treat it with suspicion, and eventually disdain. They lose trust in the people building it, and won’t recommend it.

In contrast, pleasant software suggests care and attention, and earns loyalty. Beneath the surface, it is assumed, yet more care is lurking. Decently designed products feel more trustworthy.

You can spend a lifetime learning design, but you can learn how to spot jankiness right now.

> ###  **Make friends with the grid**

Reviewer 2: ```suggestion
###  Make friends with the grid
```

Grids make things look tidy and deliberate. If something feels _vaguely janky,_ it might be because visual elements aren’t lining up according to any rules. 

> ![](https://res.cloudinary.com/dmukukwp6/image/upload/3a940cd7_5227_472c_bf55_d47193d8010d_2020x962_40f183c715.png)

Reviewer 2: In dark mode, you can't see the text on this image sadly. Maybe needs a separate version

Scattered elements laying everywhere are janky. Even aligning things by their leading or trailing edges can get you pretty far out of jank debt.

> ###  **Be disciplined in your typography**

Reviewer 2: ```suggestion
### Be disciplined in your typography
```

![](https://res.cloudinary.com/dmukukwp6/image/upload/6119a619_3e6a_47c8_a218_b525db48f8cc_1458x706_edb5d2b620.png)

Instead of going wild with multiple typefaces, try using a single typeface with variations in weight to draw the eye towards what’s most important. This creates hierarchy, making your interface easier to scan.

![](https://res.cloudinary.com/dmukukwp6/image/upload/6d3d0326_8242_4cd1_9a15_63c5018ee3d0_1460x750_555ba64c95.png)

>  **Recommended reading:** Typography is a huge subject, [but here’s a great start](https://www.interaction-design.org/literature/topics/typography?srsltid=AfmBOoqjW0UJxE_dKkPai6HgXGrHXxSicAlWEu2AFIP3PPtVQNLwuggc) on the basics from the Interaction Design Foundation. [Typography in ten minutes](https://practicaltypography.com/typography-in-ten-minutes.html) from Practical Typography is an excellent read, too.

> ###  **Mind your hierarchy**

Reviewer 2: ```suggestion
### Mind your hierarchy
```

Speaking of hierarchy, where we position things communicates their importance. If something is essential for your users to see or interact with, don’t bury it.

In Stellaris, a strategy game about space exploration and expansion, resources like energy production, minerals, and food are central to every decision, so the UI places them at the top of the screen, visible at all times.

> ![](https://res.cloudinary.com/dmukukwp6/image/upload/bd7f1a9b_c1c3_40da_9a78_3e233f9326d5_2204x1920_3f630c99b6.png)

Reviewer 2: Same. In dark mode, you can't see the text on this image sadly. Maybe needs a separate version

For desktop layouts, establishing hierarchy can mean placing elements at the top of the screen like this. For mobile, fixed elements at the top or bottom of the screen are easy to see and reach. 

Just don’t gobble up too much real estate with an immovable element. No one wants to work in a tiny viewport.

> ###  **It can be hard to attach meaning to colors**

Reviewer 2: ```suggestion
### It can be hard to attach meaning to colors
```

![](https://res.cloudinary.com/dmukukwp6/image/upload/9d5ba50f_27d7_421e_8211_dee1432219ab_1456x762_11e8d37db9.png)

Be careful with colors: they are great as accents, useful for grabbing attention and placing emphasis. But using too many colors gets visually noisy.

Don’t assume your user will grasp particular _meanings_ of colors, either. It’s best to provide labels or legends when you need to use several, if those colors have distinct meanings, as in a chart or graph.

Red can be a good indicator for declining values and destructive actions, but did you know that in some cultures, red can be a color for _gains_ rather than losses? 📈

> ###  **Maintain consistency**

Reviewer 2: ```suggestion
###  Maintain consistency
```

Visual consistency can reduce the burden of understanding an interface.

Take regular inventories of the elements in your product, checking them for consistency. I’ve grabbed screenshots of every button and printed them out, confronting all at once the inevitable variation that creeps into a product.

![](https://res.cloudinary.com/dmukukwp6/image/upload/f22c918f_b00d_4612_840d_03e6916be163_4080x2392_a666e072d6.png)

_An off-the-shelf library like [shadcn](https://ui.shadcn.com/) can help you here._

It’s okay to have some variation in your product – a button overlaid on a photo really _should_ look different than a call to action – but too much variation just looks sloppy. 

Tighten up your consistency by reusing the same styles as appropriate.

* * *

> ##  **3.** Validate big swings with cheap prototypes

Reviewer 2: ```suggestion
##  3. Validate big swings with cheap prototypes
```

Why? Because once you’ve built software that even _sort of_ works, it has inertia. It becomes resistant to change, which means jankiness _can fossilize_.

Prototypes are code you have no intention of shipping, corners gleefully cut, dedicated just to proving whether an idea works well. 

They don’t even need to be code. 

Months before the iPad launched in 2010, the OmniGroup [built a prototype iPad app](https://www.omnigroup.com/blog/Designing_OmniGraphSketcher_for_the_iPad_) with nothing but pen, paper, and a 3D-printed iPad model.

> ![](https://res.cloudinary.com/dmukukwp6/image/upload/3ada65f1_e519_467e_ae41_c1276b23b03d_400x300_b71e1097ea.jpg) ![](https://res.cloudinary.com/dmukukwp6/image/upload/50d46707_2115_4154_8fb5_95631d8dceb7_400x310_a38bcf4061.jpg)

Reviewer 2: Works perfect here: https://github.com/PostHog/posthog.com/pull/10726
```suggestion
<ImageSlider>

![](https://res.cloudinary.com/dmukukwp6/image/upload/3ada65f1_e519_467e_ae41_c1276b23b03d_400x300_b71e1097ea.jpg) 
![](https://res.cloudinary.com/dmukukwp6/image/upload/50d46707_2115_4154_8fb5_95631d8dceb7_400x310_a38bcf4061.jpg)

</ImageSlider>
```

Author: after 30 minutes of trying and failing to add the image slider component I am killing one image and moving on

Reviewer 2: sorry for the nerd snipe!

Banking is a highly-regulated space, so there was a lot of red tape. Had we waited until everything was fully baked to get user feedback, we’d have spent 18 months building the wrong thing.

Instead, in about a week, I built a fully-interactive prototype of the product that we could immediately show in user research sessions. The insights we gained showed we could succeed by skipping the regulated bits altogether.

We were in the App Store within six months.

This approach is so important that Apple keeps a whole [prototyping team](https://developer.apple.com/news/?id=97dgimaa) on staff:

> “Often, we’re starting with a specific problem to solve. But sometimes we make things just because they seem interesting, and then figure out why and what they can help solve. It’s about giving ourselves space to figure out what feels great.”

Best of all, prototyping is cheaper than ever. 

Code generation tools like Cursor and Lovable make rich, complex design experiments easy to build _and_ discard.

>  **Recommended reading:** Prototyping can also help get [nosy, opinionated stakeholders off your back.](https://www.folklore.org/Calculator_Construction_Set.html?sort=date)

* * *

> ##  **4. You have to play with lots of things**

Reviewer 2: ```suggestion
##  4. You have to play with lots of things
```

You can’t be an island on this stuff. You need to feed your sense of taste by experiencing many products, all the time.

As you do so, think deeply about _why_ you like something. Copying the superficial bits of something you like isn’t enough, you have to understand why it’s worth copying. 

Keep screenshots and recordings of the products that both enrage and impress you. Share them with your teammates and talk about the feelings they evoke.

Everything is in scope here. Not just the software you use for work, but the tools you use, cereal boxes, instruction manuals. The world is full of lessons if you look.

You won’t use everything that inspires you, but feeding your imagination will help you produce work your users will love.

In the end, people embrace and elevate the products that make them feel good. The job is to learn how to channel those feelings into your own work.

 _Words by Danilo Campos, who is building [a hardware terminal for PostHog](https://x.com/posthog/status/1890442157883871282) just for the fun of it._

> * * *

Reviewer: ```suggestion
<NewsletterForm />
```
